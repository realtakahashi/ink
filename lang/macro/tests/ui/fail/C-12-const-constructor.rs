use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        const fn const_constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
