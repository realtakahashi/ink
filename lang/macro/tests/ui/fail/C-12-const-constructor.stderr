error: `const fn` is not supported for ink! functions
  --> $DIR/C-12-const-constructor.rs:10:9
   |
10 |         const fn const_constructor() -> Self {
   |         ^^^^^
